src/core/helpers/asyncStorage.ts

```diff
@@ -1,9 +1,18 @@
 import AsyncStorage from '@react-native-async-storage/async-storage';
 import CompaniDate from './dates/companiDates';
 
-const isTokenValid = (token: string | null, tokenExpireDate: string | null): boolean =>
-  !!token && !!tokenExpireDate && CompaniDate().isBefore(tokenExpireDate);
+const isTokenValid = (token: string | null, tokenExpireDate: string | null): boolean => {
+  if (!token || !tokenExpireDate) return false;
 
+  let formattedTokenExpireDate;
+  try {
+    formattedTokenExpireDate = CompaniDate(tokenExpireDate);
+  } catch { // if date format is a stringed JSDate (ex: "Tue Jan 04 2022 11:47:04 GMT+0100 (CET)")
```

Comment:
Que penses-tu ici de mettre en commentaire plutot : "code to ensure the transition of token to CompaniDate from stringedJSDate"

Reply:
outdated